src/async_client.rs

```diff
@@ -227,46 +233,46 @@ impl Drop for AsyncClient {
     }
 }
 
-async fn background_task(client: Arc<Mutex<ua::Client>>, cycle_time: time::Duration) {
+async fn background_task(client: Arc<Mutex<ua::Client>>, cycle_time: Duration) {
     log::debug!("Starting background task");
 
     let mut interval = time::interval(cycle_time);
-    // TODO: Customized MissedTickBehavior? Only Skip and Delay are suitable here.
-    interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
-    let mut last_tick;
+    // TODO: Offer customized `MissedTickBehavior`? Only `Skip` and `Delay` are suitable here as we
+    // don't want `Burst` to repeatedly and unnecessarily call `UA_Client_run_iterate()` many times
+    // in a row.
+    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
 
-    // `UA_Client_run_iterate()` must be run periodically and makes sure to
-    // maintain the connection (e.g. renew session) and run callback handlers.
+    // `UA_Client_run_iterate()` must be run periodically and makes sure to maintain the connection
+    // (e.g. renew session) and run callback handlers.
     loop {
-        log::trace!("Next iterate cycle");
-        let last_cycle = time::Instant::now();
+        // This await point is where the background task could be aborted. (The first tick finishes
+        // immediately, so there is no additional delay on the first iteration.)
+        interval.tick().await;
+        // Track time of cycle start to report missed cycles below.
+        let start_of_cycle = Instant::now();
 
-        let status_code = {
+        let status_code = ua::StatusCode::new({
             let Ok(mut client) = client.lock() else {
                 log::error!("Terminating background task: Client could not be locked");
                 return;
             };
 
-            // Timeout of 0 means we do not block here at all. We don't want to hold the
-            // mutex longer than necessary (because that would block requests from being
-            // sent out).
+            // Timeout of 0 means we do not block here at all. We don't want to hold the mutex
+            // longer than necessary (because that would block requests from being sent out).
             log::trace!("Running iterate");
             unsafe { UA_Client_run_iterate(client.as_mut_ptr(), 0) }
-        };
-        if status_code != UA_STATUSCODE_GOOD {
-            log::warn!(
-                "Terminating background task: Run iterate failed with status code {status_code}"
-            );
+        });
+        if let Err(error) = Error::verify_good(&status_code) {
+            log::error!("Terminating background task: Run iterate failed, {error}");
             return;
         }
 
-        // This await point is where the background task could be aborted.
-        last_tick = interval.tick().await;
+        let time_taken = Instant::now().duration_since(start_of_cycle);
```

Comment:
`cycle_duration = start_of_cycle.elapsed()`? 

We could also add a trace log with the actual cycle duration in the `else` arm if no warning is logged.